app/controllers/users_controller.rb

```diff
@@ -42,7 +42,7 @@ def create
   end
 
   def update
-    if current_user
+    if current_user     # Is this really needed? See before filters above.
```

Comment:
I think this is taken care of by the before filter on line 3.

Reply:
@sagarpreet-chadha please let me know your thoughts on this one.

Reply:
I think it is important in fact i think it should be : 
`if current_user.email == params[:user][:email]` 
because if someone sends a **PATCH** request to **/users/update** and somehow sets the value of `current_user` then he/she can change the bio of any user . I am kinda new to Web Security and apologies if it may sound stupid :) 
What do you think @seafr ?

Reply:
I don't think they can set the value of `current_user` in this case. They can't tamper with the `session`. What you said is true of `params` though.
Let's see what @jywarren has to say about this.

PS I'm not too familiar with web security either.

Reply:
Oh you are right , I didn't see the code below...one cannot tamper the session . Makes sense  :smile: !

Reply:
So what do you think, should it be removed?

Reply:
This kind of makes the code more readable from the perspective of a first-timer .  What do you think ?